test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestHeadersTimeoutTests.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class RequestHeadersTimeoutTests
+    {
+        private static readonly TimeSpan RequestHeadersTimeout = TimeSpan.FromSeconds(10);
+        private static readonly TimeSpan LongDelay = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan ShortDelay = TimeSpan.FromSeconds(LongDelay.TotalSeconds / 10);
+
+
+        [Fact]
+        public async Task TestRequestHeadersTimeout()
+        {
+            using (var server = CreateServer())
+            {
+                var tasks = new[]
```

Comment:
I assume the purpose is to parallelize multiple tests in the same class, which xUnit doesn't support.  If so, I would split these into multiple classes (perhaps with a common base class) and let xUnit handle the parallelism, rather than explicitly running them in parallel this way.